internal/characters/kirara/skill.go

```diff
@@ -0,0 +1,205 @@
+package kirara
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/avatar"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/geometry"
+	"github.com/genshinsim/gcsim/pkg/core/targets"
+)
+
+// based on sayu/thoma frames
+// TODO: update frames
+var (
+	skillPressFrames []int
+	skillHoldFrames  []int
+)
+
+const (
+	skillPressHitmark = 11
+
+	skillHoldCDStart     = 648
+	skillHoldKickHitmark = 667
+
+	kickParticleICDKey = "kirara-kick-particle-icd"
+	rollParticleICDKey = "kirara-roll-particle-icd"
+)
+
+func init() {
+	// Tap E
+	skillPressFrames = frames.InitAbilSlice(46)
+	skillPressFrames[action.ActionDash] = 32
+	skillPressFrames[action.ActionJump] = 32
+	skillPressFrames[action.ActionSwap] = 44
+
+	// Hold E
+	skillHoldFrames = frames.InitAbilSlice(709) // Hold E -> N1/Q
+	skillHoldFrames[action.ActionDash] = 708    // Hold E -> J
+	skillHoldFrames[action.ActionJump] = 708    // Hold E -> J
+	skillHoldFrames[action.ActionSwap] = 706    // Hold E -> Swap
+}
+
+func (c *char) Skill(p map[string]int) action.ActionInfo {
+	hold := p["hold"]
+	if hold > 0 {
+		if hold > 10*60 {
+			hold = 10 * 60
+		}
+		return c.skillHold(p, hold)
+	}
+	return c.skillPress(p)
+}
+
+func (c *char) kickParticleCB(a combat.AttackCB) {
+	if a.Target.Type() != targets.TargettableEnemy {
+		return
+	}
+	if c.StatusIsActive(kickParticleICDKey) {
+		return
+	}
+	c.AddStatus(kickParticleICDKey, 0.5*60, true)
+	c.Core.QueueParticle("kirara-kick", 3, attributes.Dendro, c.ParticleDelay)
+}
+
+func (c *char) rollParticleCB(a combat.AttackCB) {
+	if a.Target.Type() != targets.TargettableEnemy {
+		return
+	}
+	if c.StatusIsActive(rollParticleICDKey) {
+		return
+	}
+	c.AddStatus(rollParticleICDKey, 4*60, true)
+	c.Core.QueueParticle("kirara-roll", 1, attributes.Dendro, c.ParticleDelay)
+}
+
+func (c *char) skillPress(p map[string]int) action.ActionInfo {
+	ai := combat.AttackInfo{
+		ActorIndex:         c.Index,
+		Abil:               "Tail-Flicking Flying Kick",
+		AttackTag:          attacks.AttackTagElementalArt,
+		ICDTag:             attacks.ICDTagNone,
+		ICDGroup:           attacks.ICDGroupDefault,
+		StrikeType:         attacks.StrikeTypeSlash,
+		Element:            attributes.Dendro,
+		Durability:         25,
+		Mult:               skillPress[c.TalentLvlSkill()],
+		HitlagHaltFrames:   0.06 * 60,
+		HitlagFactor:       0.01,
+		CanBeDefenseHalted: true,
+	}
+
+	c.Core.QueueAttack(
+		ai,
+		combat.NewCircleHitOnTargetFanAngle(c.Core.Combat.Player(), geometry.Point{Y: 1}, 3, 270),
```

Comment:
Should be `combat.NewCircleHitOnTargetFanAngle(c.Core.Combat.Player(), geometry.Point{Y: 0.5}, 2.5, 270)`.